users/noroadsleft/noroadsleft.c

```diff
@@ -0,0 +1,150 @@
+                    SEND_STRING(SS_LGUI("a"));
+                } else {
+                    SEND_STRING(SS_LCTL("a"));
+                }
+            }
+            return false;
+        case M_UNDO:
+            if (record->event.pressed) {
+                if ( macroMode == 1 ) {
+                    if ( get_mods() & MOD_MASK_SHIFT ) {
+                        SEND_STRING(SS_LSFT(SS_LGUI("z")));
+                    } else {
+                        SEND_STRING(SS_LGUI("z"));
+                    }
+                } else {
+                    SEND_STRING(SS_LCTL("z"));
+                }
+            }
+            return false;
+        case M_CUT:
+            if (record->event.pressed) {
+                if ( macroMode == 1 ) {
+                    SEND_STRING(SS_LGUI("x"));
+                } else {
+                    SEND_STRING(SS_LCTL("x"));
+                }
+            }
+            return false;
+        case M_COPY:
+            if (record->event.pressed) {
+                if ( macroMode == 1 ) {
+                    SEND_STRING(SS_LGUI("c"));
+                } else {
+                    SEND_STRING(SS_LCTL("c"));
+                }
+            }
+            return false;
+        case M_PASTE:
+            if (record->event.pressed) {
+                if ( macroMode == 1 ) {
+                    if ( get_mods() & MOD_MASK_SHIFT ) {
+                        SEND_STRING(SS_LSFT(SS_LALT(SS_LGUI("v"))));
+                    } else {
+                        SEND_STRING(SS_LGUI("v"));
+                    }
+                } else {
+                    SEND_STRING(SS_LCTL("v"));
+                }
+            }
+            return false;
+        case M_MDSWP:
+            if (record->event.pressed) {
+                macroMode ^= 1;
+            }
+            return false;
+        case KC_1 ... KC_0:
+            if (record->event.pressed) {
+                if (get_mods() & MOD_MASK_RALT) {
+                    register_code(keycode + 0x3B);
+                } else {
+                    register_code(keycode);
+                }
+            } else {
+                if (get_mods() & MOD_MASK_RALT) {
+                    unregister_code(keycode + 0x3B);
+                } else {
+                    unregister_code(keycode);
+                }
+            }
+            return false;
+        case KC_F1 ... KC_F12:
+            if (record->event.pressed) {
+                if (get_mods() & MOD_MASK_RALT) {
+                    register_code(keycode + 0x2E);
+                } else {
+                    register_code(keycode);
+                }
+            } else {
+                if (get_mods() & MOD_MASK_RALT) {
+                    unregister_code(keycode + 0x2E);
+                } else {
+                    unregister_code(keycode);
+                }
+            }
+            return false;
+    }  // switch()
+    return process_record_keymap(keycode, record);
```

Comment:
Personally, I like having this checked at top, before the switch.  Eg:

```c 
if (!process_record_keymap(keycode, record)) return false;
```
It's a bit more inline with the core.